### PR TITLE
Add bagua support for CUDA 11.6 images

### PR DESCRIPTION
**Pull request by wangraying**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/issues/14347 and https://github.com/Lightning-AI/lightning/issues/14346, remove special handings for bagua in https://github.com/Lightning-AI/lightning/pull/14348 and https://github.com/Lightning-AI/lightning/pull/14380

### Does your PR introduce any breaking changes? If yes, please list them.

No

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a]  Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a]  Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [n/a]  Did you list all the **breaking changes** introduced by this pull request?
- [n/a]  Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

File: .azure/gpu-tests.yml [hunk 1]
```diff
@@ -74,11 +74,11 @@ jobs:
         python -c "fname = 'requirements/pytorch/strategies.txt' ; lines = [line for line in open(fname).readlines() if 'bagua' not in line] ; open(fname, 'w').writelines(lines)"
         TORCH_VERSION=$(python -c "import torch; print(torch.__version__.split('+')[0])")
         CUDA_VERSION_MM=$(python -c "import torch ; print(''.join(map(str, torch.version.cuda.split('.')[:2])))")
-        CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [115,113,111,102] if $CUDA_VERSION_MM >= ver][0])")
+        CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,115,113,111,102] if $CUDA_VERSION_MM >= ver][0])")
```
Comment (akihironitta) [inline .azure/gpu-tests.yml:77]: I think we can let 115 go since it was a workaround for 116. cc @otaj
```suggestion
        CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,113,111,102] if $CUDA_VERSION_MM >= ver][0])")
```

File: dockers/base-conda/Dockerfile [hunk 2]
```diff
@@ -143,8 +143,8 @@ RUN \
 RUN \
     # install Bagua
     CUDA_VERSION_MM=$(python -c "print(''.join('$CUDA_VERSION'.split('.')[:2]))") && \
-    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [115,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
-    pip install "bagua-cuda$CUDA_VERSION_BAGUA==0.9.0" && \
+    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,115,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
```
Comment (akihironitta) [inline dockers/base-conda/Dockerfile:146]: Same as above.
```suggestion
    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
```

**Review by akihironitta — APPROVED** [review]

File: dockers/base-cuda/Dockerfile [hunk 3]
```diff
@@ -131,8 +131,8 @@ RUN \
 RUN \
     # install Bagua
     CUDA_VERSION_MM=$(python -c "print(''.join('$CUDA_VERSION'.split('.')[:2]))") && \
-    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [115,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
-    pip install "bagua-cuda$CUDA_VERSION_BAGUA==0.9.0" && \
+    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,115,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
```
Comment (akihironitta) [inline dockers/base-cuda/Dockerfile:134]:
```suggestion
    CUDA_VERSION_BAGUA=$(python -c "print([ver for ver in [116,113,111,102] if $CUDA_VERSION_MM >= ver][0])") && \
```

**Review by akihironitta — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]
@otaj can you look? I remember you made some updates before due to the ci failures.

**Comment by wangraying** [comment]
we're going to release bagua 0.9.1 in the next few days. Can change to `pip install bagua==0.9.1` after that.

**Review by Borda — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

**Comment by Borda** [comment]
@wangraying are you supporting the new PT 1.13?

**Comment by wangraying** [comment]
Thanks for your reminding, Borda. We will support PT 1.13. cc @woqidaideshi

**Comment by Borda** [comment]
> Thanks for your reminding, Borda. We will support PT 1.13. cc @woqidaideshi

hi, how is it going, any ETA? :otter:

**Comment by wangraying** [comment]
> > Thanks for your reminding, Borda. We will support PT 1.13. cc @woqidaideshi
> 
> hi, how is it going, any ETA? 🦦

Within this month.
